piperider_cli/cli.py

```diff
@@ -441,8 +442,26 @@ def compare_with_recipe(**kwargs):
 
     ret = 0
     try:
-        RecipeExecutor.exec(recipe_name=recipe, debug=debug)
-        CompareReport.exec(a=None, b=None, last=True, datasource=None,
+        recipe_config: RecipeConfiguration = RecipeExecutor.exec(recipe_name=recipe, debug=debug)
+        last = False
+        base = target = None
+        if not recipe_config.base.is_file_specified() and not recipe_config.target.is_file_specified():
+            last = True
+        else:
+            base = recipe_config.base.get_run_report()
+            target = recipe_config.target.get_run_report()
+
+            files = []
+            if not os.path.isfile(base):
```

Comment:
```

File not found: /Users/qrtt1/temp/git-repo-analytics/.piperider/outputs/git_repo_analytics-20230411154814/srun.json
```

I tried to put a non-existing file to the configuration, It raises the `file not found` is right. However, it would be better fast-fail before other commands executed.